apps/chat/src/components/Chat/RenameConversationModal.tsx

```diff
@@ -0,0 +1,231 @@
+import Modal from '../Common/Modal';
+
+export const RenameConversationModal = () => {
+  const renamingConversation = useAppSelector(
+    ConversationsSelectors.selectRenamingConversation,
+  );
+  if (renamingConversation) {
+    return <RenameConversationView />;
+  }
+};
+
+const RenameConversationView = () => {
+  const { t } = useTranslation(Translation.Chat);
+  const dispatch = useAppDispatch();
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const allConversations = useAppSelector(
+    ConversationsSelectors.selectConversations,
+  );
+
+  const renamingConversation = useAppSelector(
+    ConversationsSelectors.selectRenamingConversation,
+  );
+
+  const [newConversationName, setNewConversationName] = useState('');
+  const [isConfirmRenaming, setIsConfirmRenaming] = useState(false);
+  const [originConversationName, setOriginConversationName] = useState('');
+
+  useEffect(() => {
+    if (renamingConversation) {
+      setNewConversationName(renamingConversation.name || '');
+      setOriginConversationName(renamingConversation.name || '');
+      setTimeout(() => {
+        inputRef.current?.focus();
+        inputRef.current?.select();
+      });
+    } else {
+      setNewConversationName('');
+      setOriginConversationName('');
+      setIsConfirmRenaming(false);
+    }
+  }, [renamingConversation]);
+
+  const newName = useMemo(
+    () => prepareEntityName(newConversationName, { forRenaming: true }),
+    [newConversationName],
+  );
+
+  const performRename = useCallback(
+    (name: string) => {
+      if (!name.trim()) return;
+      if (name.length > 0 && renamingConversation) {
+        dispatch(
+          ConversationsActions.updateConversation({
+            id: renamingConversation.id,
+            values: {
+              name,
+              isNameChanged: true,
+              isShared: false,
+            },
+          }),
+        );
+        dispatch(ConversationsActions.setRenamingConversation(null));
+      }
+    },
+    [renamingConversation, dispatch],
+  );
+
+  const handleRename = useCallback(() => {
+    if (!renamingConversation) return;
+
+    if (
+      !isEntityNameOnSameLevelUnique(
+        newName,
+        renamingConversation,
+        allConversations,
+      )
+    ) {
+      dispatch(
+        UIActions.showErrorToast(
+          t(
+            'Conversation with name "{{newName}}" already exists in this folder.',
+            {
+              ns: 'chat',
+              newName,
+            },
+          ),
+        ),
+      );
+
+      return;
+    }
+
+    if (doesHaveDotsInTheEnd(newName)) {
+      dispatch(
+        UIActions.showErrorToast(
+          t('Using a dot at the end of a name is not permitted.'),
+        ),
+      );
+      return;
+    }
+
+    if (
+      renamingConversation.isShared &&
+      newName !== renamingConversation.name
+    ) {
+      setIsConfirmRenaming(true);
+      return;
+    }
+
+    performRename(trimEndDots(newName));
```

Comment:
do you really need `trimEndDots()`. You have `if (doesHaveDotsInTheEnd(newName))` statement above